pkg/codegen/nodejs/gen.go

```diff
@@ -2505,6 +2505,17 @@ func genNPMPackageMetadata(pkg *schema.Package, info NodePackageInfo, localDepen
 		npminfo.Resolutions[resk] = resv
 	}
 
+	for _, dep := range pkg.Dependencies {
+		if npminfo.Dependencies == nil {
```

Comment:
Dependencies are full package specs which means you can grab the plugin, and get _it's_ schema and so correctly resolve the NPM name even taking into account namespaces and language overrides.